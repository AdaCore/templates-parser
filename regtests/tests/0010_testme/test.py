from test_support import *

run('testme', ["testme16.tmplt"])
run('print_tree', ["testme16.tmplt"])
