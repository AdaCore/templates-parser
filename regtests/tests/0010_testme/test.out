
   OK 2

   OK 2

   OK 2

   OK

   OK

   OK

   OK

   OK

   OK

   OK

   OK

   OK

   OK

   OK

[INFO] testme16.tmplt
[C_INFO]  1 FALSE
[TEXT] 
[IF] (@_TWO_@ = 2)
  [TEXT]    OK 2
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] (@_TWO_@ /= 3)
  [TEXT]    OK 2
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] (not (@_TWO_@ = 3))
  [TEXT]    OK 2
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] ((@_ONE_@ = 1) or (not (@_TWO_@ = 3)))
  [TEXT]    OK
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] ((@_ONE_@ = 2) or (not (@_TWO_@ = 3)))
  [TEXT]    OK
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] ((@_ONE_@ = 2) and (not (@_TWO_@ = 3)))
  [TEXT]    NOK
[ELSE]
  [TEXT]    OK
[END_IF]
[TEXT] 
[IF] ((@_ONE_@ = 1) and (not (@_TWO_@ = 3)))
  [TEXT]    OK
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] ((not (@_TWO_@ = 3)) and (not (@_ONE_@ = 2)))
  [TEXT]    OK
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] ((not (@_TWO_@ = 3)) and (not (@_ONE_@ = 2)))
  [TEXT]    OK
[ELSE]
  [TEXT]    NOK
[END_IF]
[TEXT] 
[IF] ((not (@_TWO_@ = 3)) and (not (@_ONE_@ = 1)))
  [TEXT]    NOK
[ELSE]
  [TEXT]    OK
[END_IF]
[TEXT] 
[IF] ((not (@_TWO_@ = 3)) and (not (@_ONE_@ = 1)))
  [TEXT]    NOK
[ELSE]
  [TEXT]    OK
[END_IF]
[TEXT] 
[IF] (not @_COND2_@)
  [TEXT]    NOK
[ELSE]
  [TEXT]    OK
[END_IF]
[TEXT] 
[IF] ((@_ONE_@ = 1) and (not @_COND2_@))
  [TEXT]    NOK
[ELSE]
  [TEXT]    OK
[END_IF]
[TEXT] 
[IF] ((@_ONE_@ = 1) and (not @_COND2_@))
  [TEXT]    NOK
[ELSE]
  [TEXT]    OK
[END_IF]
